### ISSUE TITLE
Document tracing internals

### ISSUE THREAD

**Issue by gvanrossum** [python/cpython#96815]
Let's add a new .md file to the Python directory explaining how tracing is done in the interpreter.

Brandt wrote it up [here](https://github.com/python/cpython/issues/96636#issuecomment-1241330947).

**Comment by brandtbucher** [comment]
Yep, I agree this is good to record alongside the code. I've been working on a draft that fleshes out the comment you linked to (should be done soon).

**Comment by anujrmohite** [comment]
Hey @brandtbucher ,

Just checking - still is this open, finishing the tracing docs draft?
I'm ready to invest time understanding and contributing to this
as it would help me understand CPython internals further. 
Or no problem if you have it covered! Thanks!

**Comment by gvanrossum** [comment]
I'm not Brandt, but the implementation has changed quite a bit with the implementation of [PEP 669](https://peps.python.org/pep-0669/) (low-impact monitoring, usually referred to as "instrumentation" in the code). It would still be useful to document the implementation, but I believe @markshannon is the only one who could truly review such documentation.

**Comment by anujrmohite** [comment]
sure, thanks for the context.
